lib/helpers.rb

```diff
@@ -160,6 +161,8 @@ def handle_threads_query(
       end
     end
 
+    comment_threads = comment_threads.where({"context" => context})
```

Comment:
It depends on whether it changes how Mongo chooses/uses the index.  You may want to add this field to a new composite index for efficient retrieval, but I would run a quick loadtest to see if there's any difference first.  We should still have a usable JMeter test; DM if you need details.